heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java

```diff
@@ -152,6 +152,9 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
     // deactivate and sleep
     if (initiallyRunning) {
       deactivateTopology(stateManager, updatedTopology);
+      // Update the topology since the state should have changed from RUNNING to PAUSED
+      updatedTopology =
```

Comment:
Ah right, we'd have to pass in proposedPackingPlan to the deactivateTopology method. That's unfortunate, but I think that's our best option. That way we can at least encapsulate the deactivateTopology logic to return a new valid topology that aligns with the one passed in.